internal/serve/middleware/middleware.go

```diff
@@ -285,16 +285,31 @@ func CSPMiddleware() func(http.Handler) http.Handler {
 
 // ResolveTenantFromRequestMiddleware is a middleware that injects the tenant into the request context, if it can be found in
 // the request HEADER, or the hostname prefix.
-func ResolveTenantFromRequestMiddleware(tenantManager tenant.ManagerInterface) func(http.Handler) http.Handler {
+func ResolveTenantFromRequestMiddleware(tenantManager tenant.ManagerInterface, singleTenantMode bool) func(http.Handler) http.Handler {
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
 			ctx := req.Context()
 
 			var currentTenant *tenant.Tenant
-
-			// Attempt fetching tenant name from request
-			if tenantName, err := extractTenantNameFromRequest(req); err == nil && tenantName != "" {
-				currentTenant, _ = tenantManager.GetTenantByName(ctx, tenantName)
+			if singleTenantMode {
+				var err error
+				currentTenant, err = tenantManager.GetDefault(ctx)
+				if err != nil {
+					switch {
+					case errors.Is(err, tenant.ErrTenantDoesNotExist):
+						httperror.Unauthorized("No default Tenant configured", err, nil).Render(rw)
+					case errors.Is(err, tenant.ErrTooManyDefaultTenants):
+						httperror.Unauthorized("Too many default tenants configured", err, nil).Render(rw)
+					default:
+						httperror.InternalError(ctx, "", fmt.Errorf("error getting default tenant: %w", err), nil).Render(rw)
+					}
+					return
+				}
```

Comment:
4xx are usually caused by a problem with the request/client, while 5xx refers to problems in the server and server data.

In this case, I don't see why we would return Unauthorized. a 500 for any of these cases would make more sense to me since there's nothing the client can do to get a 2xx response.

Reply:
I'll change them to 500 but I'll keep the messages separated to make the debug easier